### DISCUSSION TITLE
Optional field on @@unique with default

### DISCUSSION THREAD

**Discussion by leppaott** [prisma/prisma#17359] [Q&A]
If we have such model, without any PKs:
```
model Model {
  time   DateTime @db.Timestamptz
  serial BigInt   @default(autoincrement())

  @@unique([time, serial])
}
```

Is it possible that we change the `serial` to be optional and always use the default value (serial incrementing)?
Trying to come up with Timescaledb workaround as we don't need any PKs/Unique indexes this is only to satisfy Prisma requirements.

**Comment by leppaott** [comment]
Maybe a better example would be such as:

```
model Model {
  uuid String @default(uuid()) @db.Uuid
  email String

  @@unique([uuid, email])
}
```

We can clearly see that uuid, is always defaulted using `uuid` function. So my thinking is that we should be able to set uuid to be optional:

```
model Model {
  uuid String? @default(uuid()) @db.Uuid
  email String

  @@unique([uuid, email])
}
```

But we can't due to unique indexing. I'd like to drop uuid from all methods and always to use the default.

**Reply by nurul3101** [reply]
Hey @leppaott 

This is a valid model which would allow you to mark uuid as optional

```prisma
model Model {
  uuid  String? @default(uuid()) @db.Uuid
  email String

  @@unique([email])
}
```

Would this work for you? Each model must have at least one unique criteria that have only required fields, and in this case it is fulfilled by email field.

**Reply by leppaott** [reply]
Yes that is a workaround if your model has another non-optional field.

In essence for a single @unique field.
```
model Model {
  uuid String? @unique @default(uuid()) @db.Uuid
}
```

I believe SQL database doesn't care in this case whether the `CREATE UNIQUE INDEX` column is `NOT NULL`

In my first case with `serial`, the serial needs to be part of the unique index so can't just use the `time` or some other field.

Similarly for two @@unique fields:

```
model Model {
  uuid String? @default(uuid()) @db.Uuid
  uuidSecond String? @default(uuid()) @db.Uuid

  @@unique([uuid, uuidSecond])
}
```

I would expect this to work where @default() is clearly used and there is no chance that uuids are NULL.
Could be that NULLs could be supported but this might change on database...

**Reply by nurul3101** [reply]
Yes, for now, the fields that are passed while defining the unique index should not be marked as optional even if there is a default attribute assigned to those optional fields.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
